File: javascript/Baekjoon/Bronze/V/25304.js
// 영수증
// https://www.acmicpc.net/problem/25304

let input = require('fs').readFileSync('../../예제.txt').toString().replace(/\r*/g, "").trim().split('\n');

let money = Number(input.shift());
let count = Number(input.shift());
let sum = 0;

for(let i = 0; i < count; i++) {
    let nums = input[i].split(' ').map(Number);
    sum += nums[0] * nums[1];
}

console.log(money == sum ? 'Yes' : 'No');

File: Python/beakjoon/Bronze/V/25304.py
# 영수증
# https://www.acmicpc.net/problem/25304

money = int(input())
count = int(input())
data = [list(map(int, input().split())) for _ in range(count)]
result = 0

for i in data:
    result += i[0] * i[1]

if money == result:
    print('Yes')
else:
    print('No')